source/math/FloatMatrix.ooc

```diff
@@ -123,12 +126,13 @@ FloatMatrix : cover {
 	// Tranpose matrix. Creates a new matrix being the transpose of the current matrix.
 	// </summary>
 	// <returns>Return current matrix tranposed.</returns>
-	transpose: func@ -> This {
+	transpose: func -> This {
 		result := This new(this dimensions swap())
 		for (y in 0 .. this height)
 			for (x in 0 .. this width)
 				result elements[y + x * this height] = this elements[x + y * this width]
-		result
+		this free(Owner Callee)
```

Comment:
We're calling `free~withCriteria()`, so it will be freed if the ownership is set to `Owner Callee`, i.e. if the instance was not explicitly taken with `take()` or directly instantiated via a constructor which defaults owner to `Owner Caller`.
```ooc
free: func@ ~withCriteria (criteria: Owner) -> Bool {
	this _owner == criteria && this free()
}
```